x/distribution/keeper/allocation.go

```diff
@@ -52,34 +53,52 @@ func (k Keeper) AllocateTokens(ctx context.Context, totalPreviousPower int64, bo
 	feeMultiplier := feesCollected.MulDecTruncate(voteMultiplier)
 
 	// allocate tokens proportionally to voting power
-	//
-	// TODO: Consider parallelizing later
-	//
-	// Ref: https://github.com/cosmos/cosmos-sdk/pull/3099#discussion_r246276376
+	var g errgroup.Group
+	// channel for remaining rewards
+	remainingCh := make(chan sdk.DecCoins, len(bondedVotes))
+
 	for _, vote := range bondedVotes {
+		vote := vote
 
-		validator, err := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
-		if err != nil {
-			return err
-		}
+		g.Go(func() error {
+			validator, err := k.stakingKeeper.ValidatorByConsAddr(ctx, vote.Validator.Address)
+			if err != nil {
+				return err
+			}
 
-		// TODO: Consider micro-slashing for missing votes.
-		//
-		// Ref: https://github.com/cosmos/cosmos-sdk/issues/2525#issuecomment-430838701
-		powerFraction := math.LegacyNewDec(vote.Validator.Power).QuoTruncate(math.LegacyNewDec(totalPreviousPower))
-		reward := feeMultiplier.MulDecTruncate(powerFraction)
+			powerFraction := math.LegacyNewDec(vote.Validator.Power).QuoTruncate(math.LegacyNewDec(totalPreviousPower))
+			reward := feeMultiplier.MulDecTruncate(powerFraction)
 
-		err = k.AllocateTokensToValidator(ctx, validator, reward)
-		if err != nil {
-			return err
-		}
+			if err = k.AllocateTokensToValidator(ctx, validator, reward); err != nil {
+				return err
+			}
+
+			remainingCh <- reward
+			return nil
+		})
+	}
+
+	if err := g.Wait(); err != nil {
+		return err
+	}
+	close(remainingCh)
 
+	// calculate the remaining rewards
+	for reward := range remainingCh {
 		remaining = remaining.Sub(reward)
 	}
 
-	// allocate community funding
-	feePool.CommunityPool = feePool.CommunityPool.Add(remaining...)
-	return k.FeePool.Set(ctx, feePool)
+	// send to community pool and set remainder in fee pool
+	amt, re := remaining.TruncateDecimal()
+	if err := k.bankKeeper.SendCoinsFromModuleToModule(ctx, types.ModuleName, protocolpooltypes.ModuleName, amt); err != nil {
```

Comment:
can we set the protocolpool name in dist to avoid the dep, unless we are importing it elsewhere

Reply:
Will remove, but we need protocolpool for simulations